src/display/node_stream.js

```diff
@@ -30,6 +30,10 @@ const fileUriRegex = /^file:\/\/\/[a-zA-Z]:\//;
 
 class PDFNodeStream {
   constructor(source) {
+    // prepend 'file:///' to windows absolute path
+    if (source.url.match(/^[a-z]:[/\\]/i)) {
+      source.url = `file:///${source.url}`;
+    }
```

Comment:
Don't overwrite the `source.url` variable, use a separate variable instead. It might be a good idea to add a helper function that runs your logic, and then calls `url.parse`. Then replace the `url.parse` call below with a call to your function.